localstack/services/cloudformation/engine/entities.py

```diff
@@ -164,10 +164,25 @@ def set_stack_status(self, status: str, status_reason: Optional[str] = None):
         self.metadata["StackStatus"] = status
         if "FAILED" in status:
             self.metadata["StackStatusReason"] = status_reason or "Deployment failed"
+        self.log_stack_errors()
         self.add_stack_event(
             self.stack_name, self.stack_id, status, status_reason=status_reason or ""
         )
 
+    def log_stack_errors(self, level=logging.WARNING):
+        for event in self.events:
+            if event["ResourceStatus"].endswith("FAILED"):
+                if reason := event.get("ResourceStatusReason"):
+                    reason = reason.replace("\n", "; ")
+                    LOG.log(
+                        level,
+                        "CFn resource failed to deploy: %s (%s)",
+                        event["LogicalResourceId"],
+                        reason,
+                    )
+                else:
+                    LOG.warning("CFn resource failed to deploy: %s", event["LogicalResourceId"])
```

Comment:
this else branch should (ideally) never happen though, right? :thinking:

Reply:
One would hope... unfortunately it looks like `set_resource_status` calls into `add_stack_event` (which constructs the stack event, and takes `status_reason` as its argument), and this is called in one place with `*_FAILED`

https://github.com/localstack/localstack/blob/46d2dae7fb393977e73b8ce670c27a5b80ac0c91/localstack/services/cloudformation/engine/template_deployer.py#L1376-L1384

I will take a look to see if I can rectify this as part of this PR